File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort()
        prev = points[0][1]
        cnt = 1
        n= len(points)
        for i in range(1,n):
            if points[i][0] <= prev :
                prev = min(prev, points[i][1])
            else:
                prev = points[i][1]
                cnt+=1
        return cnt

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(),points.end());
        int prev = points[0][1];
        int cnt = 1, n= points.size();
        for(int i=1;i<n;i++){
            if(points[i][0] <= prev ){
                prev = min(prev, points[i][1]);
            }else{
                prev = points[i][1];
                cnt+=1;
            }
        }
        return cnt;
    }
};